### PR TITLE
Sync Service

### PR DESCRIPTION
**Pull request by blainsmith**
This is a basic sync service within the same process as the server than you can configure to run on a certain interval. This is not ideal for production, but illustrations a method to get data in a database from 3rd party services.

Perhaps a future plan is to break out this logic into another repo, but Oddworks does not care how videos get into the database and we also don't want to re-invent the wheels of scheduled jobs since there are many projects out there that do just that. Maybe we break out the individual provider scripts, but at the end of the day they are really just REST clients hitting an API and transforming data into Odd schemas. Fairly trivial.

### CODE REVIEW

**Comment by pmcconnell** [comment]
looks good. some thoughts...

- should we have the loading of the sync service be optional? do we always want/need it loaded.  add env var to prevent loading of sync service?

- consider allowing setting the interval externally as well.

- we should look at a command line interface to run the syncService so we can manually sync and not have it always running. handy for local/test dev work

- I don't have a problem with the sync service being in the main repo for now but the vimeo provider should be off loaded sooner than later.

I'll shut up now. Ship it. 👍

**Comment by blainsmith** [comment]
1. It is optional. Just delete the line all together. `server.js` is merely a recommended setup and we cannot have the "kitchen sink" be configurable without it getting severely complicated. Anyone who runs this SHOULD edit server.js for their own stores and services they need.

2. That is also up to the dev as well who runs this. Tho certainly a recommended setting to. Ideally this should NOT be run in production as is.

3. This can't happen until we can another Oddcast transport in place since they are 2 different processes. I am trying to get a TCP Transport working for this.

4. The entire service should be yes. This is just a slim implementation. Ideally someone should use something like https://www.npmjs.com/package/kue or https://www.npmjs.com/package/cron or whatever you feel like doing to run the logic of the vimeo file.

**Comment by brkattk** [comment]
I think this would conflict with #48 - if that goes first I would be happy to get this working

**Comment by blainsmith** [comment]
Cool, merge away then. It was more of a proof of concepts, but ideally in prod it would run elsewhere. Not to say someone couldn't run a second Oddworks instance with JUST sync running and remove the API.

TL;DR lets merge this.

**Comment by brkattk** [comment]
Opted to leave sync service out of the default config, but left it commented for an example
